src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetSampleAnnotationFmsDTOValidator.java

```diff
@@ -0,0 +1,292 @@
+		}
+
+		if (StringUtils.isNotEmpty(dto.getAbundance())) {
+			htpSampleAnnotation.setAbundance(dto.getAbundance());
+		}
+		
+		if (StringUtils.isNotEmpty(dto.getSampleType())) {
+			String curie = dto.getSampleType();
+			OBITerm obiTerm = obiTermService.findByCurie(curie);
+			if (obiTerm != null) {
+				htpSampleAnnotation.setHtpExpressionSampleType(obiTerm);
+			} else {
+				htpSampleAnnotationResponse.addErrorMessage("SampleType", ValidationConstants.INVALID_MESSAGE + " (" + curie + ")");
+			}
+		} else {
+			htpSampleAnnotationResponse.addErrorMessage("SampleType", ValidationConstants.REQUIRED_MESSAGE);
+		}
+
+		if (dto.getSampleAge() != null) {
+			if (htpSampleAnnotation.getHtpExpressionSampleAge() == null) {
+				htpSampleAnnotation.setHtpExpressionSampleAge(new BioSampleAge());
+			}
+			ObjectResponse<TemporalContext> temporalContextObjectResponse = geneExpressionAnnotationFmsDTOValidator.validateTemporalContext(dto.getSampleAge().getStage());
+			if (temporalContextObjectResponse.hasErrors()) {
+				htpSampleAnnotationResponse.addErrorMessage("Sample Age - Stage", temporalContextObjectResponse.errorMessagesString());
+			} else {
+				TemporalContext temporalContext = geneExpressionAnnotationFmsDTOValidator.updateTemporalContext(temporalContextObjectResponse, htpSampleAnnotation.getHtpExpressionSampleAge().getStage());
+				htpSampleAnnotation.getHtpExpressionSampleAge().setAge(dto.getSampleAge().getAge());
+				htpSampleAnnotation.getHtpExpressionSampleAge().setWhenExpressedStageName(dto.getSampleAge().getStage().getStageName());
+				htpSampleAnnotation.getHtpExpressionSampleAge().setStage(temporalContext);
+			}
+		}
+
+		List<Long> idsToRemove = new ArrayList<>();
+		if (CollectionUtils.isNotEmpty(dto.getSampleLocations())) {
+			if (CollectionUtils.isNotEmpty(htpSampleAnnotation.getHtpExpressionSampleLocations())) {
+				for (AnatomicalSite anatomicalSite : htpSampleAnnotation.getHtpExpressionSampleLocations()) {
+					idsToRemove.add(anatomicalSite.getId());
+				}
+			}
+			List<AnatomicalSite> htpSampleLocations = new ArrayList<>();
+			for (WhereExpressedFmsDTO whereExpressedDTO : dto.getSampleLocations()) {
+				ObjectResponse<AnatomicalSite> anatomicalSiteObjectResponse = geneExpressionAnnotationFmsDTOValidator.validateAnatomicalSite(whereExpressedDTO);
+				if (anatomicalSiteObjectResponse.hasErrors()) {
+					htpSampleAnnotationResponse.addErrorMessage("SampleLocations", anatomicalSiteObjectResponse.errorMessagesString());
+				} else {
+					htpSampleLocations.add(anatomicalSiteObjectResponse.getEntity());
+				}
+			}
+			htpSampleAnnotation.setHtpExpressionSampleLocations(htpSampleLocations);
+		}
+
+		if (dto.getGenomicInformation() != null) {
+			Boolean genomicInformationExists = (StringUtils.isNotEmpty(dto.getGenomicInformation().getBioSampleText()) == true || StringUtils.isNotEmpty(dto.getGenomicInformation().getBiosampleId()) == true);
```

Comment:
🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.coding.SimplifyBooleanExpressionCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Expression can be simplified.
<!-- __reviewdog__:ChBlMDdmOTE4Njg5MTNkMTgxEgpjaGVja3N0eWxl -->